tests/Aspire.Hosting.Redis.Tests/RedisFunctionalTests.cs

```diff
@@ -0,0 +1,227 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using StackExchange.Redis;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Aspire.Hosting.Redis.Tests;
+
+public class RedisFunctionalTests
+{
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyRedisResource()
+    {
+        var builder = TestDistributedApplicationBuilder.Create();
+
+        var redis = builder.AddRedis("redis");
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{redis.Resource.Name}"] = await redis.Resource.GetConnectionStringAsync()
+        });
+
+        hb.AddRedisClient(redis.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        var redisClient = host.Services.GetRequiredService<IConnectionMultiplexer>();
+
+        var db = redisClient.GetDatabase();
+
+        await db.StringSetAsync("key", "value");
+
+        var value = await db.StringGetAsync("key");
+
+        Assert.Equal("value", value);
+    }
+
+    [Fact(Skip = "Debugging")]
+    [RequiresDocker]
+    public async Task WithDataVolumeShouldPersistStateBetweenUsages()
+    {
+        var volumeName = "myvolume1";
+
+        // Use a volume to do a snapshot save
+
+        await VerifyDataPersistence(
+            options => options.WithDataVolume(volumeName),
+            async redisClient => await redisClient.GetServers().First().SaveAsync(SaveType.BackgroundSave)
+            );
+    }
+
+    [Fact(Skip = "Debugging")]
+    [RequiresDocker]
+    public async Task WithDataVolumeWithCustomPersistenceInterval()
+    {
+        var volumeName = "myvolume2";
+        var snapshotInterval = TimeSpan.FromSeconds(1);
+
+        // Use a volume to do a snapshot save with a custom interval
+
+        await VerifyDataPersistence(
+            options => options.WithDataVolume(volumeName).WithPersistence(snapshotInterval),
+            async redisClient => await Task.Delay(snapshotInterval + TimeSpan.FromSeconds(1))
```

Comment:
This seems like it will be flaky - especially on slow machines.

Reply:
I am ok to remove it, as long as we agree that the other explicit SAVE call is sufficient to test the functionality since we already have a unit test that ensures the correct `--save` argument is passed with a custom timespan. This test is only validating that the feature actually works in Redis. Or we can add more that "1 more second after the time that Redis should have saved it".